File: customer-service/src/main/java/vnlaw/service/customerservice/dto/response/ApiResponse.java
package vnlaw.service.customerservice.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
public class ApiResponse <T> {
    int code;
    String message;
    T result;

}

File: front-end/mobile/lib/api_service/api_response.dart
class ApiResponse<T> {
  int? httpCode;
  bool? result;
  String? message;
  T? data;
  T? error;

  ApiResponse(
      {this.httpCode, this.result, this.message, this.data, this.error});

  @override
  String toString() {
    return '''
  ApiResponse {
    httpCode: $httpCode
    result: $result
    message: $message
    data: $data
    error: $error
  }''';
  }
}
